Fix port handling

jmstover opened the pull request:
Looking at the port value was using `ParseInt()` which used a
`int16` value. Change to `ParseUint()` to cover the full range we
require with a `uint16` value.

Fixes: #4028 

Attn: @singularity-maintainers

jmstover committed:
Fix port handling

Looking at the port value was using `ParseInt()` which used a
`int16` value. Change to `ParseUint()` to cover the full range we
require with a `uint16` value.

jscook2345 committed:
Merge branch 'master' into port_fix

cclerget reviewed (commented):
Looks good to me, thanks @jmstover . Do you plan to include unit tests in this PR or in a separated one ?

mike-dubman commented on pkg/network/network_linux.go:
```diff
@@ -283,7 +283,7 @@ func (m *Setup) SetArgs(args []string) error {
 					return fmt.Errorf("portmap port argument is badly formatted")
 				}
-				if n, err := strconv.ParseInt(ports[0], 0, 16); err == nil {
+				if n, err := strconv.ParseUint(ports[0], 0, 16); err == nil {
```
looks good, line 288 and 297 cannot happen now, i guess.

jmstover replied:
I left those in for now, so that just incase the `ParseUint()` gets reverted somehow, we still have an error check in place. Though... I should make another small change.

sashayakovtseva replied:
imo useless checks should be removed in this PR.
How `ParseUint()` can be reverted?

mike-dubman replied:
or better check that port is in range 0-65k and fail if not. If something gets reverted - it will still be correct.

jmstover replied:
I was going to make it a `uint16` across the board since that fits the port range nicely. But in the Golang unix.SockaddrInet4 function, the port is an *int* value. So... that just breaks that idea, so I need to keep letting it use the lines like:
```
pm.HostPort = int(n)
```

As for getting reverting, it's _possible_ that a merge could have the old `ParseInt()` code. So, when it gets merged, it reverts it back to the incorrect way. Made more likely if someone force pushes.

jmstover committed:
Add test

Add simple test to check that a valid uint16 port passes, and an
invalid port fails.

jmstover committed:
Add max port value check

jmstover committed:
Merge branch 'master' into port_fix

jmstover committed:
Merge branch 'master' into port_fix

jscook2345 committed:
Merge branch 'master' into port_fix

jscook2345 reviewed (approved):
🥇 @jmstover !
